src/models/sunburst.js

```diff
@@ -235,15 +235,15 @@ nv.models.sunburst = function() {
 
             cGE.append("path")
                 .attr("d", arc)
-                .style("fill", function (d) {
+                .style("fill", function (d, i) {
                     if (d.color) {
                         return d.color;
                     }
                     else if (groupColorByParent) {
-                        return color((d.children ? d : d.parent).name);
+                        return color((d.children ? d : d.parent), i);
                     }
                     else {
-                        return color(d.name);
+                        return color(d, i);
```

Comment:
The default color function is nv.utils.defaultColor, which uses nv.utils.getColor which has a definition of just function(color).  While it might work, it's going to hash a whole object every time, which is going to end up generating different colors if the data structure is modified.  

Why would you be changing/duplicating the name attribute anyway?  If you really have to do that, then I suggest putting a unique number on the end of the name and then strip it off with a custom tooltip.headerFormatter option